spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/client/HttpClientBeanPostProcessorTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.web.client;
+
+import brave.propagation.TraceContext;
+import io.netty.bootstrap.Bootstrap;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+
+import org.springframework.cloud.sleuth.instrument.web.client.HttpClientBeanPostProcessor.PendingClientSpan;
+import org.springframework.cloud.sleuth.instrument.web.client.HttpClientBeanPostProcessor.TracingMapConnect;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpClientBeanPostProcessorTest {
```

Comment:
no good deed goes unpunished :) @marcingrzejszczak any idea why this dies in maven, but not in intellij?

Reply:
Ordering maybe? Delay? The question is why it's failing.

Reply:
both assertions failed and only thing I can think of is if somehow reactor subscriber context has static or thread local state which can taint other things.. I pulled this file out for now.